exonum/src/blockchain/mod.rs

```diff
@@ -481,6 +481,10 @@ impl Blockchain {
                 // Consensus messages cache is useful only during one height, so it should be
                 // cleared when a new height is achieved.
                 schema.consensus_messages_cache_mut().clear();
+                let txs_in_block = schema.last_block().tx_count();
```

Comment:
Actually I don't understand how transaction count from last block relates to size of pool after commit. I'd suggest that this value should be something like this:
```rust
let tx_pool_size_after_commit = {
       let txs_count = schema.transactions_pool_len_index().get().unwrap_or(0);
       if txs_count <= txs_block_limit {
              0
       else {
              txs_count - txs_block_limit
       }
};
```

Reply:
As I understand we assume here that all transactions included in the last block were present in our pool, so in order to get accurate pool entry count we should subtract the transaction count in the last block from our pool size cache.

However, it is possible that the last committed block originates from other node which had a different pool state? For example, our pool has [T1, T2], other node's pool has [T1, T2, T3, T4], the block includes [T1, T2, T3, T4]. In this case it seems we'll still have an overflow on line 487.

Reply:
The question is: Which exactly last block do we get in this case ? I suppose it will be a block with current height but we need to know a size of block that will be generated after `self.merge(patch)?;` don't we ?

Reply:
>The question is: Which exactly last block do we get in this case ? I suppose it will be a block with current height but we need to know a size of block that will be generated after self.merge(patch)?; don't we ?

We are working with Fork that contain's "new block". So in `fork` last_block is really block at current height. But in db\blockchain\node_state `last_block` is still outdated block.

>Actually I don't understand how transaction count from last block relates to size of pool after commit. I'd suggest that this value should be something like this:
```rust
let tx_pool_size_after_commit = {
       let txs_count = schema.transactions_pool_len_index().get().unwrap_or(0);
       if txs_count <= txs_block_limit {
              0
       else {
              txs_count - txs_block_limit
       }
};
```

I don't get it. If we get `txs_count > txs_block_limit` it's a bug, because we assume that block's tx is always in the pool.